ingestion/src/metadata/data_insight/processor/web_analytic_report_data_processor.py

```diff
@@ -0,0 +1,339 @@
+#  Copyright 2021 Collate
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""
+Processor class used to compute refined report data
+"""
+
+from __future__ import annotations
+
+from collections import namedtuple
+from datetime import datetime, timedelta
+from typing import Generator, Iterable, Optional
+
+from metadata.data_insight.processor.data_processor import DataProcessor
+from metadata.generated.schema.analytics.reportData import ReportData, ReportDataType
+from metadata.generated.schema.analytics.reportDataType.webAnalyticEntityViewReportData import (
+    WebAnalyticEntityViewReportData,
+)
+from metadata.generated.schema.analytics.reportDataType.webAnalyticUserActivityReportData import (
+    WebAnalyticUserActivityReportData,
+)
+from metadata.generated.schema.analytics.webAnalyticEventData import (
+    WebAnalyticEventData,
+)
+from metadata.generated.schema.entity.data import (
+    chart,
+    dashboard,
+    database,
+    databaseSchema,
+    mlmodel,
+    pipeline,
+    table,
+    topic,
+)
+from metadata.generated.schema.entity.teams import user
+from metadata.generated.schema.entity.teams.user import User
+from metadata.generated.schema.tests import testSuite
+from metadata.ingestion.api.source import SourceStatus
+from metadata.utils.helpers import get_entity_tier_from_tags
+from metadata.utils.logger import data_insight_logger
+
+logger = data_insight_logger()
+
+LIMIT = 1000
+ENTITIES = {
+    "chart": chart.Chart,
+    "dashboard": dashboard.Dashboard,
+    "database": database.Database,
+    "databaseSchema": databaseSchema.DatabaseSchema,
+    "mlmodel": mlmodel.MlModel,
+    "pipeline": pipeline.Pipeline,
+    "table": table.Table,
+    "topic": topic.Topic,
+    "user": user.User,
+    "testSuite": testSuite.TestSuite,
+}
+
+CACHED_EVENTS = []
+START_TS = str(
+    int(
+        (datetime.utcnow() - timedelta(days=0))
+        .replace(hour=0, minute=0, second=0, microsecond=0)
+        .timestamp()
```

Comment:
nit: I think this can be simplified with `datetime.combine(datetime.utcnow(), time.min).timestamp()`